packages/express-did-auth/src/classes/session-manager.ts

```diff
@@ -0,0 +1,81 @@
+import { SessionManager, UserSessionConfig } from '../types'
+import { randomBytes } from 'crypto'
+import { ErrorCodes } from '../errors'
+import { DEFAULT_USER_SESSION_DURATION } from '../constants'
+
+export interface UserSessionInfo {
+  did: string
+  expirationDate: number
+  metadata: any
+}
+
+export interface UserSessionDictionary {
+  [refreshToken: string]: UserSessionInfo
+}
+
+export interface DidRefreshTokenMapping {
+  [did: string]: string
+}
+
+export default class implements SessionManager {
+  private sessionDurationInHours: number
+  private userSessions: UserSessionDictionary
+  private didRefreshTokenMapping: DidRefreshTokenMapping
+
+  constructor ({ userSessionDurationInHours }: UserSessionConfig) {
+    this.sessionDurationInHours = userSessionDurationInHours || DEFAULT_USER_SESSION_DURATION
+    this.userSessions = {}
+    this.didRefreshTokenMapping = {}
+  }
+
+  create (did: string, metadata?: any): string {
+    if (!did) throw new Error(ErrorCodes.INVALID_DID)
+
+    // invalidates prior token
+    const oldRefreshToken = this.didRefreshTokenMapping[did]
+    if (oldRefreshToken) delete this.userSessions[oldRefreshToken]
+
+    const refreshToken = randomBytes(64).toString('hex')
+
+    this.userSessions[refreshToken] = {
+      did,
+      expirationDate: Date.now() + this.sessionDurationInHours * 60 * 60 * 1000,
```

Comment:
Agree

Reply:
I think in this context is good to have the time unit expressed in the name of the variable, it helps to prevent misunderstandings. Take in account that JWT times are expressed in seconds (check `NumericDate ` subtitle in https://tools.ietf.org/html/rfc7519#section-2 ) and `Date.now()` returns milisecs. If we include the unit in the variable name, then it is easier to know what to do prior to each comparison.

Reply:
I believe this is black box stuff. Developers will not have option to import this class from the lib index